awx_collection/plugins/modules/tower_instance_group.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+# coding: utf-8 -*-
+
+
+# (c) 2020, John Westcott IV <john.westcott.iv@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_instance_group
+author: "John Westcott IV (@john-westcott-iv)"
+version_added: "4.0"
+short_description: create, update, or destroy Ansible Tower instance groups.
+description:
+    - Create, update, or destroy Ansible Tower instance groups. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name of this instance group.
+      required: True
+      type: str
+    new_name:
+      description:
+        - Setting this option will change the existing name (looked up via the name field.
+      required: True
+      type: str
+    credential:
+      description:
+        - Credential to authenticate with Kubernetes or OpenShift.  Must be of type "Kubernetes/OpenShift API Bearer Token”.
+      required: False
+      type: str
+    policy_instance_percentage:
+      description:
+        - Minimum percentage of all instances that will be automatically assigned to this group when new instances come online.
+      required: False
+      type: int
+      default: '0'
+    policy_instance_minimum:
+      description:
+        - Static minimum number of Instances that will be automatically assign to this group when new instances come online.
+      required: False
+      type: int
+      default: '0'
+    policy_instance_list:
+      description:
+        - List of exact-match Instances that will be assigned to this group
+      required: False
+      type: dict
```

Comment:
why is this a dict, as opposed to a list?

Reply:
That came from the API:
```
            "policy_instance_list": {
                "type": "json",
                "required": false,
                "label": "Policy Instance List",
                "help_text": "List of exact-match Instances that will be assigned to this group",
                "filterable": true,
                "child": {
                    "type": "string",
                    "required": true,
                    "read_only": false
                }
            },
```

Reply:
ah, so yeah, I think that's because the API doesn't actually have a list type. My concern here is whether this is actually a usable field. The API really needs a list, but I don't know for sure that Ansible itself will allow a list in a dict field. I might see if I can suggest an example for this real quick and also determine if it does or doesn't work.

Reply:
Yeah, this is what I hit

```
TASK [Create an instance group with list] *******************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "argument policy_instance_list is of type <class 'list'> and we were unable to convert to dict: <class 'list'> cannot be converted to a dict"}
```

from

```yaml
    - name: Create an instance group with list
      tower_instance_group:
        name: "alan instance group"
        policy_instance_list:
          - awx
```

I think that the easy solution here is to make it the a list type. Tested with

```diff
diff --git a/awx_collection/plugins/modules/tower_instance_group.py b/awx_collection/plugins/modules/tower_instance_group.py
index 6424be4b3b..9a7db2f814 100644
--- a/awx_collection/plugins/modules/tower_instance_group.py
+++ b/awx_collection/plugins/modules/tower_instance_group.py
@@ -54,7 +54,7 @@ options:
       description:
         - List of exact-match Instances that will be assigned to this group
       required: False
-      type: dict
+      type: list
     pod_spec_override:
       description:
         - A custom Kubernetes or OpenShift Pod specification.
@@ -88,7 +88,7 @@ def main():
         credential=dict(),
         policy_instance_percentage=dict(type='int', default='0'),
         policy_instance_minimum=dict(type='int', default='0'),
-        policy_instance_list=dict(type='dict'),
+        policy_instance_list=dict(type='list'),
         pod_spec_override=dict(),
         instances=dict(required=False, type="list", default=None),
         state=dict(choices=['present', 'absent'], default='present'),
```

and that seems to work fine.